File: Factory Method/FactoryMethod - Java/principal/src/main/java/com/formatfactory/pagamento/model/BoletoBancario.java
package com.formatfactory.pagamento.model;

import org.springframework.stereotype.Component;

@Component("BoletoBancario") 
public class BoletoBancario implements Pagamento {
    @Override
    public void realizaPagamento(Double valor) {
        System.out.println("Pagando " + valor + " usando boleto bancário.");
    }    
}

File: Factory Method/FactoryMethod - PHP/src/model/BoletoBancario.php
<?php

require_once "Pagamento.php";

class BoletoBancario implements Pagamento{
    public function realizaPagamento(Float $valor){
        echo "Pagamento de " . number_format($valor, 2, ',', '.') . " realizado utilizando boleto bancário.";
    }
}

?>